taichi/ir/frontend_ir.cpp

```diff
@@ -1124,8 +1124,8 @@ void GetElementExpression::type_check(CompileConfig *config) {
   TI_ASSERT_TYPE_CHECKED(src);
   auto func_call = src.cast<FuncCallExpression>();
   TI_ASSERT(func_call);
-  TI_ASSERT(index < func_call->func->rets.size());
-  ret_type = func_call->func->rets[index].dt;
+  TI_ASSERT(index[0] < func_call->func->rets.size());
```

Comment:
This seems a bit confusing, what is `index[0]` and what's special about it?

Reply:
Should probably put a comment somewhere denoting that `index[0]` is different

Reply:
The return value is still flattened now, so the length of the index is always 1. I will refactor in the subsequent PRs to let it compile to nested structs.

Reply:
In fact they are refactored in #7027 lol